### PR TITLE
add nightwatch

### PR DESCRIPTION
**Pull request by markwolff**
- Add nightwatch e2e testing
- Supports testing for `ie8`, `ie9`, `ie10`, `ie11`, `edge`, `firefox`, `chrome`

Run with `npm run nightwatch`
  - This runs (in parallel) tests for ie10, ie11, chrome. Firefox and edge cannot be run concurrently

Run a browser test separately
`npm run nightwatch:edge`
`npm run nightwatch:firefox`

### CODE REVIEW

File: AISKU/Tests/nightwatch/run_nightwatch.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+module.exports = {
+  '1. Navigate to tests' : function (browser) {
+    var reportName = `tests_output/qunit_output_${browser.options.desiredCapabilities.browserName}${browser.options.desiredCapabilities.version || ''}.html`;
+    var ieAppendage = browser.options.desiredCapabilities.version;
+    ieAppendage = ieAppendage ? `?version=${ieAppendage}` : '';
+    browser
+      .url(`http://localhost:8000/Tests/Selenium/Tests.html${ieAppendage}`)
```
Comment (jpiyali) [inline AISKU/Tests/nightwatch/run_nightwatch.js]: why is this needed?

Comment (markwolff) [reply AISKU/Tests/nightwatch/run_nightwatch.js]: I add a meta tag to `ie` tests so that I can emulate other version of ie. This is so I can grab the version config option I set in `nightwatch.json` and pass it along to `serve_nightwatch.js`

https://docs.microsoft.com/en-us/windows/desktop/win7appqual/use-the-meta-tag-to-ensure-future-compatibility
